src/backend/chat/commands/command-handler.ts

```diff
@@ -0,0 +1,640 @@
+
+        frontendCommunicator.send("active-cooldowns-reset");
+    }
+
+    clearCooldownsForSingleCommand(commandId: string): void {
+        const keys = this._cooldownCache.keys();
+        keys.forEach((k) => {
+            if (k.split(CACHE_KEY_SEPARATOR)[0] === commandId) {
+                this._cooldownCache.del(k);
+            }
+        });
+
+        frontendCommunicator.send("cooldowns-cleared-for-command", commandId);
+    }
+
+    private buildCommandRegexStr(trigger: string, scanWholeMessage: boolean): string {
+        const escapedTrigger = util.escapeRegExp(trigger);
+        if (scanWholeMessage) {
+            return `(?:^|\\s)${escapedTrigger}(?!-)(?:\\b|$|(?=\\s))`;
+        }
+        return `^${escapedTrigger}(?!-)(?:\\b|$|(?=\\s))`;
+    }
+
+    private testForTrigger(message: string, trigger: string, scanWholeMessage: boolean, triggerIsRegex: boolean): boolean {
+        message = message.toLowerCase();
+
+        const normalizedTrigger = trigger.toLowerCase();
+        const commandRegexStr = triggerIsRegex
+            ? trigger
+            : this.buildCommandRegexStr(normalizedTrigger, scanWholeMessage);
+
+        const regex = new RegExp(commandRegexStr, "gi");
+
+        return regex.test(message);
+    }
+
+    private checkForCommand(rawMessage: string): CommandMatch {
+        if (rawMessage == null || rawMessage.length < 1) {
+            return null;
+        }
+
+        const allCommands = commandManager.getAllActiveCommands();
+
+        for (const command of allCommands) {
+
+            if (this.testForTrigger(
+                rawMessage,
+                command.trigger,
+                command.scanWholeMessage,
+                command.triggerIsRegex
+            )
+            ) {
+                return { command, matchedTrigger: command.trigger };
+            }
+
+            if (!command.triggerIsRegex && command.aliases != null && Array.isArray(command.aliases)) {
+                for (const alias of command.aliases) {
+                    if (this.testForTrigger(
+                        rawMessage,
+                        alias,
+                        command.scanWholeMessage,
+                        false
+                    )) {
+                        return { command, matchedTrigger: alias };
+                    }
+                }
+            }
+        }
+        return { command: null };
+    }
+
+    private updateCommandCount(command: CommandDefinition): void {
+        if (command.count == null) {
+            command.count = 0;
+        }
+        command.count++;
+        frontendCommunicator.send("command-count-update", {
+            commandId: command.id,
+            count: command.count
+        });
+    }
+
+    private parseCommandTriggerAndArgs(trigger: string, rawMessage: string, scanWholeMessage = false, treatQuotedTextAsSingleArg = false): TriggerWithArgs {
+        let args = [], rawArgs = [];
+
+        if (rawMessage != null) {
```

Comment:
Super minor (and probably preexisting but just trying to be thorough): since`rawArgs` isn't needed in the scope its defined in, it can be moved within the `if(rawMessage != null) { }` block.